# Analysis on directors

#### In this notebook, we analyze directors and their gender distribution

### Loading CMU Corpus

In [2]:
CMU_PATH = "data/MovieSummaries"
IMDB_PATH = "data/imdb_data"

In [4]:
import ast
import time
import random
import logging
import pandas as pd
import numpy as np
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from urllib.error import HTTPError
import plotly.express as px
from tqdm import tqdm
from joblib import Parallel, delayed
from multiprocessing import Pool
from SPARQLWrapper import SPARQLWrapper, JSON
from src.utils.utils_directors import process_batch_with_backoff, fetch_gender_wikidata, process_gender_parallel

We load the csv we created from CMU data and imdb

In [5]:
metadata = pd.read_csv(f"data/merged_data.csv")

In [12]:
metadata

Wikipedia movie ID Freebase_movie_ID                Movie_name  \
0                  975900         /m/03vyhn            Ghosts of Mars   
1                28463795        /m/0crgdbh               Brun bitter   
2                 9363483        /m/0285_cd          White Of The Eye   
3                  261236         /m/01mrr1         A Woman in Flames   
4                10408933        /m/02qc0j7  Alexander's Ragtime Band   
...                   ...               ...                       ...   
57117            23851782        /m/06_vb43           The Ghost Train   
57118            32468537        /m/0crwd9y           Shadow Boxing 2   
57119            34980460        /m/0g4pl34                   Knuckle   
57120             9971909        /m/02pygw1         Another Nice Mess   
57121            12476867        /m/02w7zz8                   Spliced   

       Movie_release_date_x  Movie_box_office_revenue  Movie_runtime  \
0                      2001                14010832.0           98.0   
1                      1988                       NaN           83.0   
2                      1987                       NaN          110.0   
3                      1983                       NaN          106.0   
4                      1938                 3600000.0          106.0   
...                     ...                       ...            ...   
57117                  1941                       NaN           82.0   
57118                  2007                       NaN          132.0   
57119                  2011                       NaN           96.0   
57120                  1972                       NaN           66.0   
57121                  2002                       NaN           86.0   

                                Movie_languages  \
0                          ['English Language']   
1                        ['Norwegian Language']   
2                          ['English Language']   
3                           ['German Language']   
4                          ['English Language']   
...                                         ...   
57117                      ['English Language']   
57118  ['Russian Language', 'English Language']   
57119                      ['English Language']   
57120                      ['English Language']   
57121                      ['English Language']   

                     Movie_countries  \
0       ['United States of America']   
1                         ['Norway']   
2                 ['United Kingdom']   
3                        ['Germany']   
4       ['United States of America']   
...                              ...   
57117             ['United Kingdom']   
57118                     ['Russia']   
57119  ['Ireland', 'United Kingdom']   
57120   ['United States of America']   
57121                     ['Canada']   

                                            Movie_genres     tconst  ...  \
0      ['Thriller', 'Science Fiction', 'Horror', 'Adv...  tt0228333  ...   
1                             ['Crime Fiction', 'Drama']  tt0094806  ...   
2      ['Thriller', 'Erotic thriller', 'Psychological...  tt0094320  ...   
3                                              ['Drama']  tt0083949  ...   
4               ['Musical', 'Comedy', 'Black-and-white']  tt0029852  ...   
...                                                  ...        ...  ...   
57117  ['Crime Fiction', 'Thriller', 'Comedy', 'Super...  tt0033660  ...   
57118     ['Sports', 'Crime Fiction', 'Action', 'Drama']  tt0459759  ...   
57119      ['Biographical film', 'Drama', 'Documentary']  tt1606259  ...   
57120                               ['Satire', 'Comedy']  tt0362411  ...   
57121          ['Thriller', 'Horror', 'Slasher', 'Teen']  tt0354216  ...   

                                                  gender  \
0      ['F', 'M', 'F', 'M', 'F', 'F', 'M', 'F', 'M', ...   
1      ['M', 'F', 'F', 'M', 'M', 'M', 'F', 'M', 'M', ...   
2      ['M', 'F', 'M', 'M', 'M', 'F', 'F', 'M', 'M', ...   
3      ['F', 'M', 

We load the crew dataset from idmb

In [13]:
crew = pd.read_table(f"{IMDB_PATH}/title.crew.tsv")

crew

tconst   directors  \
0         tt0000001   nm0005690   
1         tt0000002   nm0721526   
2         tt0000003   nm0721526   
3         tt0000004   nm0721526   
4         tt0000005   nm0005690   
...             ...         ...   
10625491  tt9916848   nm1485677   
10625492  tt9916850   nm1485677   
10625493  tt9916852   nm1485677   
10625494  tt9916856  nm10538645   
10625495  tt9916880   nm0996406   

                                                    writers  
0                                                        \N  
1                                                        \N  
2                                                        \N  
3                                                        \N  
4                                                        \N  
...                                                     ...  
10625491  nm9187127,nm1485677,nm9826385,nm9299459,nm1628284  
10625492            nm9187127,nm1485677,nm9826385,nm1628284  
10625493  nm9187127,nm1485677,nm9826385,nm9299459,nm1628284  
10625494                                          nm6951431  
10625495                                nm1482639,nm2586970  

[10625496 rows x 3 columns]

We merge our crew dataset with metadata

In [14]:
crew = crew.merge(metadata, on="tconst", how="inner")

In [15]:
crew

tconst            directors                           writers  \
0      tt0000591            nm0141150                         nm0141150   
1      tt0000615            nm0533958               nm0092809,nm0533958   
2      tt0000679  nm0091767,nm0877783               nm0000875,nm0877783   
3      tt0001101            nm0923603                                \N   
4      tt0001341            nm1024681                         nm1024681   
...          ...                  ...                               ...   
57117  tt9815072            nm0445390                         nm0445390   
57118  tt9834564            nm0353721                         nm1834334   
57119  tt9855214            nm0539034  nm10511503,nm10511505,nm10511504   
57120  tt9881364            nm7843279     nm7843279,nm0707494,nm2270180   
57121  tt9898504            nm0634777                                \N   

       Wikipedia movie ID Freebase_movie_ID                      Movie_name  \
0                 4849466        /m/07s67rr               L'Enfant prodigue   
1                32986669        /m/04p7yxx              Robbery Under Arms   
2                 5954041         /m/0fgfyy  The Fairylogue and Radio-Plays   
3                 6334750         /m/0g1kr9      Abraham Lincoln's Clemency   
4                28149517        /m/0cmbf1t      Jarní sen starého mládence   
...                   ...               ...                             ...   
57117            18475018        /m/04f2nb9              Pontianak Menjerit   
57118            22780907        /m/05zwjv0              Oshidori kenkagasa   
57119             7316179         /m/0kv4t6                           Kisan   
57120            17475562        /m/04g0_80                            Gaja   
57121             8783319        /m/027jl28            Puteri Gunong Ledang   

       Movie_release_date_x  Movie_box_office_revenue  Movie_runtime  \
0                      1907                       NaN           90.0   
1                      1907                       NaN            NaN   
2                      1908                       NaN          120.0   
3                      1910                       NaN            NaN   
4                      1913                       NaN            NaN   
...                     ...                       ...            ...   
57117                  2005                       NaN          105.0   
57118                  1957                       NaN            NaN   
57119                  2006                       NaN            NaN   
57120                  2008                       NaN            NaN   
57121                  1961                       NaN            NaN   

                           Movie_languages  ...  \
0                      ['French Language']  ...   
1                          ['Silent film']  ...   
2                     ['English Language']  ...   
3                     ['English Language']  ...   
4                       ['Czech Language']  ...   
...                                    ...  ...   
57117  ['Malay Language', 'Thai Language']  ...   
57118                ['Japanese Language']  ...   
57119               ['Malayalam Language']  ...   
57120                                   []  ...   
57121                   ['Malay Language']  ...   

                                                  gender  \
0                                   ['M', 'M', 'F', 'F']   
1                         ['M', 'M', 'M', 'M', 'M', 'F']   
2      ['M', 'M', 'M', 'M', 'M', 'M', 'M', 'M', 'M', ...   
3                                                  ['M']   
4                         ['M', 'M', 'F', 'F', 'M', 'M']   
...                                                  ...   
57117  ['F', 'M', 'M', 'F', 'M', 'M', 'F', 'M', 'F', ...   
57118  ['M', 'F', 'M', 'M', 'M', 'M', 'F', 'M', 'M', ...   
57119  ['M', 'M', 'F', 'F', 'F', 'M', 'M', 'M', 'F', ...   
57120  ['F', 'F', 'M', 'F', 'M', 'F', 'F', 'F', 'M', ...   
57121      ['F', 'F', 

We replace the "\N" character by NaN in the directors column to make the procesing easier

In [16]:
crew['directors'] = crew['directors'].replace("\\N", np.nan)

In [17]:
crew[crew["directors"].isnull()]

tconst directors              writers  Wikipedia movie ID  \
109    tt0004667       NaN                   \N            35261321   
222    tt0006249       NaN                   \N            35261105   
306    tt0006996       NaN  nm0535508,nm0000636            15097511   
569    tt0008984       NaN            nm0235323            13436659   
12230  tt0051719       NaN                   \N            20123164   
...          ...       ...                  ...                 ...   
56911  tt6548518       NaN                   \N            26208838   
56967  tt6925244       NaN                   \N            30415996   
56968  tt6925264       NaN                   \N            33270310   
56990  tt7450784       NaN                   \N            23519606   
56998  tt7624456       NaN                   \N            12932687   

      Freebase_movie_ID                                  Movie_name  \
109          /m/0j7lhp1                           Taking his Chance   
222          /m/0j7kryj  We'll Take Her Children in Amongst Our Own   
306          /m/03hh9c3                                     Macbeth   
569          /m/03c591c                               Cupid Angling   
12230        /m/04y7qh2                            The Hidden World   
...                 ...                                         ...   
56911        /m/0b77vwp                               Thodi Kodallu   
56967        /m/0g5sf7q                                    Graduate   
56968        /m/0h7r1h6                                    Kshetram   
56990        /m/06w9_rf                        Vedikkai En Vadikkai   
56998        /m/02z0f12                             Kolomba Sanniya   

       Movie_release_date_x  Movie_box_office_revenue  Movie_runtime  \
109                    1914                       NaN            NaN   
222                    1915                       NaN            NaN   
306                    1915                       NaN            NaN   
569                    1918                       NaN            NaN   
12230                  1958                       NaN            NaN   
...                     ...                       ...            ...   
56911                  1957                       NaN            NaN   
56967                  2011                       NaN            NaN   
56968                     0                       NaN            NaN   
56990                  1990                       NaN            NaN   
56998                  1976                       NaN            NaN   

                           Movie_languages  ...  \
109    ['Silent film', 'English Language']  ...   
222    ['Silent film', 'English Language']  ...   
306                    ['French Language']  ...   
569                   ['English Language']  ...   
12230                 ['English Language']  ...   
...                                    ...  ...   
56911                  ['Telugu language']  ...   
56967                  ['Telugu language']  ...   
56968                                   []  ...   
56990                   ['Tamil Language']  ...   
56998                 ['Sinhala Language']  ...   

                                                  gender  \
109                                                ['F']   
222                                                ['F']   
306                                 ['M', 'F', 'M', 'M']   
569                                           ['M', 'F']   
12230                                              ['M']   
...                                                  ...   
56911                                              ['F']   
56967                                         ['M', 'M']   
56968                                    ['M', 'M', 'M']   
56990  ['M', 'M', 'F', 'M', 'F', 'F', 'F', 'M', 'F', ...   
56998  ['M', 'M', 'M', 'M', 'F', 'M', 'M', 'M', 'F', ...   

                                                  actors  first_role_actor  \
109                        

We have 130 movies with missing directors. Thus, we query wikidata to find them.

In [18]:

endpoint_url = "https://query.wikidata.org/sparql"
sparql = SPARQLWrapper(endpoint_url)

imdb_ids = crew.loc[crew['directors'].isna(), 'tconst'].tolist()

batch_size = 20  
batched_ids = [imdb_ids[i:i + batch_size] for i in range(0, len(imdb_ids), batch_size)]

failed_batches = []

for batch in batched_ids:
    print(f"Processing batch: {batch}")
    directors_data = process_batch_with_backoff(batch, sparql)
    
    for imdb_id, directors in directors_data.items():
        try:
            row_index = crew[crew['tconst'] == imdb_id].index[0]
            crew.at[row_index, 'directors'] = ", ".join(d['director'] for d in directors)
        except Exception as e:
            print(f"Error updating IMDb ID {imdb_id}: {e}")
            failed_batches.append(batch)

    time.sleep(5) 

if failed_batches:
    print("Retrying failed batches...")
    for batch in failed_batches:
        print(f"Retrying batch: {batch}")
        directors_data = process_batch_with_backoff(batch, sparql)
        for imdb_id, directors in directors_data.items():
            try:
                row_index = crew[crew['tconst'] == imdb_id].index[0]
                crew.at[row_index, 'directors'] = ", ".join(d['director'] for d in directors)
            except Exception as e:
                print(f"Failed again for IMDb ID {imdb_id}: {e}")

print("Batch processing complete!")


Processing batch: ['tt0004667', 'tt0006249', 'tt0006996', 'tt0008984', 'tt0051719', 'tt0055313', 'tt0068834', 'tt0072703', 'tt0137811', 'tt0166220', 'tt0182993', 'tt0185893', 'tt0197077', 'tt0224364', 'tt0246877', 'tt0247283', 'tt0250486', 'tt0274019', 'tt0274566', 'tt0280997']
Processing batch: ['tt0281116', 'tt0292192', 'tt0296893', 'tt0312714', 'tt0314385', 'tt0314521', 'tt0315735', 'tt0316404', 'tt0318015', 'tt0319645', 'tt0319808', 'tt0320617', 'tt0328395', 'tt0331567', 'tt0336573', 'tt0346931', 'tt0347595', 'tt0349182', 'tt0350678', 'tt0351784']
Processing batch: ['tt0353777', 'tt0361753', 'tt0370765', 'tt0371870', 'tt0372549', 'tt0373244', 'tt0377126', 'tt0381050', 'tt0385515', 'tt0402305', 'tt0438054', 'tt0483559', 'tt0497710', 'tt0887254', 'tt0904079', 'tt0997281', 'tt1059311', 'tt1067580', 'tt10681962', 'tt1279240']
Processing batch: ['tt13613000', 'tt13818954', 'tt14189060', 'tt14350400', 'tt14630666', 'tt14630674', 'tt14652506', 'tt14653694', 'tt14653732', 'tt14852902', 'tt

In [19]:
crew.directors.isnull().sum()

65

We can see that we have some missing values for the directors.

In [20]:
crew[crew.directors.isnull()]

tconst directors              writers  Wikipedia movie ID  \
306    tt0006996       NaN  nm0535508,nm0000636            15097511   
12230  tt0051719       NaN                   \N            20123164   
17633  tt0068834       NaN                   \N            18916948   
30192  tt0137811       NaN                   \N            33839507   
33059  tt0182993       NaN                   \N            21450473   
...          ...       ...                  ...                 ...   
56889  tt6448760       NaN                   \N            13180192   
56903  tt6450176       NaN                   \N            17489405   
56967  tt6925244       NaN                   \N            30415996   
56968  tt6925264       NaN                   \N            33270310   
56998  tt7624456       NaN                   \N            12932687   

      Freebase_movie_ID                           Movie_name  \
306          /m/03hh9c3                              Macbeth   
12230        /m/04y7qh2                     The Hidden World   
17633        /m/04jkrxj                      The Last Target   
30192        /m/0hn96b_                          Her Triumph   
33059        /m/05f6yb7  Wakanohana monogatari dohyou no oni   
...                 ...                                  ...   
56889        /m/02zbfx3                           Tholaipesi   
56903        /m/0479g_h                                 1977   
56967        /m/0g5sf7q                             Graduate   
56968        /m/0h7r1h6                             Kshetram   
56998        /m/02z0f12                      Kolomba Sanniya   

       Movie_release_date_x  Movie_box_office_revenue  Movie_runtime  \
306                    1915                       NaN            NaN   
12230                  1958                       NaN            NaN   
17633                     0                       NaN            NaN   
30192                  1915                       NaN            NaN   
33059                  1992                       NaN            NaN   
...                     ...                       ...            ...   
56889                  2007                       NaN            NaN   
56903                  2008                       NaN            NaN   
56967                  2011                       NaN            NaN   
56968                     0                       NaN            NaN   
56998                  1976                       NaN            NaN   

             Movie_languages  ...  \
306      ['French Language']  ...   
12230   ['English Language']  ...   
17633                     []  ...   
30192                     []  ...   
33059  ['Japanese Language']  ...   
...                      ...  ...   
56889     ['Tamil Language']  ...   
56903     ['Tamil Language']  ...   
56967    ['Telugu language']  ...   
56968                     []  ...   
56998   ['Sinhala Language']  ...   

                                                  gender  \
306                                 ['M', 'F', 'M', 'M']   
12230                                              ['M']   
17633                                              ['M']   
30192                                         ['F', 'M']   
33059                                              ['M']   
...                                                  ...   
56889                                              ['F']   
56903           ['F', 'M', 'F', 'F', 'M', 'M', 'M', 'M']   
56967                                         ['M', 'M']   
56968                                    ['M', 'M', 'M']   
56998  ['M', 'M', 'M', 'M', 'F', 'M', 'M', 'M', 'F', ...   

                                                  actors  first_role_actor  \
306    ['Séverin-Mars', 'Georgette Leblanc', 'Pierre ...      Séverin-Mars   
12230                                   ['Gregory Peck']      Gregory Peck   
17633                               ['Michael Redgrave']  Michael Redgrave   
30192                    ['Gaby Deslys', 'Har

We can see that these movies overall contain many missing values. To simplify the analysis, we decide to drop them. 

In [21]:
crew = crew.dropna(subset=["directors"])

We also notice that some movies have multiple directors. For the sake of our analysis, we explode the directors column so that we have all directors in separate rows.

In [22]:
crew['directors'] = crew['directors'].str.split(',')

crew_long = crew.explode('directors')

crew_long = crew_long.reset_index(drop=True)

C:\Users\USER\AppData\Local\Temp\ipykernel_15536\1423380569.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  crew['directors'] = crew['directors'].str.split(',')


In [23]:
crew_long

tconst  directors                           writers  \
0      tt0000591  nm0141150                         nm0141150   
1      tt0000615  nm0533958               nm0092809,nm0533958   
2      tt0000679  nm0091767               nm0000875,nm0877783   
3      tt0000679  nm0877783               nm0000875,nm0877783   
4      tt0001101  nm0923603                                \N   
...          ...        ...                               ...   
62084  tt9815072  nm0445390                         nm0445390   
62085  tt9834564  nm0353721                         nm1834334   
62086  tt9855214  nm0539034  nm10511503,nm10511505,nm10511504   
62087  tt9881364  nm7843279     nm7843279,nm0707494,nm2270180   
62088  tt9898504  nm0634777                                \N   

       Wikipedia movie ID Freebase_movie_ID                      Movie_name  \
0                 4849466        /m/07s67rr               L'Enfant prodigue   
1                32986669        /m/04p7yxx              Robbery Under Arms   
2                 5954041         /m/0fgfyy  The Fairylogue and Radio-Plays   
3                 5954041         /m/0fgfyy  The Fairylogue and Radio-Plays   
4                 6334750         /m/0g1kr9      Abraham Lincoln's Clemency   
...                   ...               ...                             ...   
62084            18475018        /m/04f2nb9              Pontianak Menjerit   
62085            22780907        /m/05zwjv0              Oshidori kenkagasa   
62086             7316179         /m/0kv4t6                           Kisan   
62087            17475562        /m/04g0_80                            Gaja   
62088             8783319        /m/027jl28            Puteri Gunong Ledang   

       Movie_release_date_x  Movie_box_office_revenue  Movie_runtime  \
0                      1907                       NaN           90.0   
1                      1907                       NaN            NaN   
2                      1908                       NaN          120.0   
3                      1908                       NaN          120.0   
4                      1910                       NaN            NaN   
...                     ...                       ...            ...   
62084                  2005                       NaN          105.0   
62085                  1957                       NaN            NaN   
62086                  2006                       NaN            NaN   
62087                  2008                       NaN            NaN   
62088                  1961                       NaN            NaN   

                           Movie_languages  ...  \
0                      ['French Language']  ...   
1                          ['Silent film']  ...   
2                     ['English Language']  ...   
3                     ['English Language']  ...   
4                     ['English Language']  ...   
...                                    ...  ...   
62084  ['Malay Language', 'Thai Language']  ...   
62085                ['Japanese Language']  ...   
62086               ['Malayalam Language']  ...   
62087                                   []  ...   
62088                   ['Malay Language']  ...   

                                                  gender  \
0                                   ['M', 'M', 'F', 'F']   
1                         ['M', 'M', 'M', 'M', 'M', 'F']   
2      ['M', 'M', 'M', 'M', 'M', 'M', 'M', 'M', 'M', ...   
3      ['M', 'M', 'M', 'M', 'M', 'M', 'M', 'M', 'M', ...   
4                                                  ['M']   
...                                                  ...   
62084  ['F', 'M', 'M', 'F', 'M', 'M', 'F', 'M', 'F', ...   
62085  ['M', 'F', 'M', 'M', 'M', 'M', 'F', 'M', 'M', ...   
62086  ['M', 'M', 'F', 'F', 'F', 'M', 'M', 'M', 'F', ...   
62087  ['F', 'F', 'M', 'F', 'M', 'F', 'F', 'F', 'M', ...   
62088      ['F', 'F', 'M', 'F', 'M', 'M', 'M', 'M', 'M']   

                                                  actors  first_role_actor  \
0  

We have some duplicates because directors generally have multiple movies. Thus, we create a dataframe that only contains a director once. 

In [24]:
crew_long_unique = crew_long.drop_duplicates(subset=["directors"])

We load the name dataset to map the crew nconst to the name

In [25]:
imdb_names = pd.read_csv(f"{IMDB_PATH}/name.basics.tsv", sep="\t", engine = "pyarrow")

In [26]:
imdb_names

nconst         primaryName birthYear deathYear  \
0         nm0000001        Fred Astaire      1899      1987   
1         nm0000002       Lauren Bacall      1924      2014   
2         nm0000003     Brigitte Bardot      1934        \N   
3         nm0000004        John Belushi      1949      1982   
4         nm0000005      Ingmar Bergman      1918      2007   
...             ...                 ...       ...       ...   
13944213  nm9993714   Romeo del Rosario        \N        \N   
13944214  nm9993716       Essias Loberg        \N        \N   
13944215  nm9993717  Harikrishnan Rajan        \N        \N   
13944216  nm9993718         Aayush Nair        \N        \N   
13944217  nm9993719          Andre Hill        \N        \N   

                            primaryProfession  \
0                actor,miscellaneous,producer   
1          actress,soundtrack,archive_footage   
2           actress,music_department,producer   
3               actor,writer,music_department   
4                       writer,director,actor   
...                                       ...   
13944213  animation_department,art_department   
13944214                                   \N   
13944215                      cinematographer   
13944216                      cinematographer   
13944217                                   \N   

                                   knownForTitles  
0         tt0050419,tt0072308,tt0053137,tt0027125  
1         tt0037382,tt0075213,tt0117057,tt0038355  
2         tt0057345,tt0049189,tt0056404,tt0054452  
3         tt0072562,tt0077975,tt0080455,tt0078723  
4         tt0050986,tt0069467,tt0050976,tt0083922  
...                                           ...  
13944213          tt11657662,tt14069590,tt2455546  
13944214                                       \N  
13944215                                tt8736744  
13944216                                tt8736744  
13944217                                       \N  

[13944218 rows x 6 columns]

In [27]:
imdb_names = imdb_names.rename(columns={"nconst": "directors"})

We merge with our dataset.

In [28]:
crew_long = crew_long.merge(imdb_names[['directors', 'primaryName', "birthYear", "deathYear"]], on="directors", how="left")


In [29]:
crew_long.head()

tconst  directors              writers  Wikipedia movie ID  \
0  tt0000591  nm0141150            nm0141150             4849466   
1  tt0000615  nm0533958  nm0092809,nm0533958            32986669   
2  tt0000679  nm0091767  nm0000875,nm0877783             5954041   
3  tt0000679  nm0877783  nm0000875,nm0877783             5954041   
4  tt0001101  nm0923603                   \N             6334750   

  Freebase_movie_ID                      Movie_name  Movie_release_date_x  \
0        /m/07s67rr               L'Enfant prodigue                  1907   
1        /m/04p7yxx              Robbery Under Arms                  1907   
2         /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
3         /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
4         /m/0g1kr9      Abraham Lincoln's Clemency                  1910   

   Movie_box_office_revenue  Movie_runtime       Movie_languages  ...  \
0                       NaN           90.0   ['French Language']  ...   
1                       NaN            NaN       ['Silent film']  ...   
2                       NaN          120.0  ['English Language']  ...   
3                       NaN          120.0  ['English Language']  ...   
4                       NaN            NaN  ['English Language']  ...   

  second_role_actor     third_role_actor  first_role_gender  \
0      Henri Gouget  Christiane Mandelys                  M   
1   George Merriman           Lance Vane                  M   
2       Frank Burns     George E. Wilson                  M   
3       Frank Burns     George E. Wilson                  M   
4               NaN                  NaN                  M   

   second_role_gender third_role_gender  F  M       primaryName birthYear  \
0                   M                 F  1  2      Michel Carré      1865   
1                   M                 M  0  3  Charles MacMahon      1861   
2                   M                 M  0  3     Francis Boggs      1870   
3                   M                 M  0  3       Otis Turner      1862   
4                 NaN               NaN  0  1  Theodore Wharton      1875   

  deathYear  
0      1945  
1      1917  
2      1911  
3      1918  
4      1931  

[5 rows x 28 columns]

In [30]:
crew_long[crew_long.primaryName.isnull()]

tconst              directors              writers  \
134    tt0004667       Raymond Longford                   \N   
270    tt0006249       Raymond Longford                   \N   
673    tt0008984  Louis William Chaudet            nm0235323   
14909  tt0055313           Milan Vošmik  nm0389348,nm0435079   
19492  tt0068846             nm16784584  nm0000419,nm0330948   
...          ...                    ...                  ...   
61849  tt6447878      A. Govindamoorthy                   \N   
61851  tt6448728                  Saran                   \N   
61855  tt6449318           A. Venkatesh                   \N   
61872  tt6548518      Adurthi Subba Rao                   \N   
61953  tt7450784                   Visu                   \N   

       Wikipedia movie ID Freebase_movie_ID  \
134              35261321        /m/0j7lhp1   
270              35261105        /m/0j7kryj   
673              13436659        /m/03c591c   
14909            28379474        /m/0crjdmc   
19492             3404048         /m/099qsr   
...                   ...               ...   
61849            23967551        /m/076xthx   
61851            16490980        /m/03y8m3g   
61855            17274239        /m/043pqjg   
61872            26208838        /m/0b77vwp   
61953            23519606        /m/06w9_rf   

                                       Movie_name  Movie_release_date_x  \
134                             Taking his Chance                  1914   
270    We'll Take Her Children in Amongst Our Own                  1915   
673                                 Cupid Angling                  1918   
14909                    Pohádka o staré tramvaji                  1961   
19492                              Letter to Jane                  1972   
...                                           ...                   ...   
61849                         Vedigundu Murugesan                  2009   
61851                              Modhi Vilayadu                  2008   
61855                                  Singakutty                  2008   
61872                               Thodi Kodallu                  1957   
61953                        Vedikkai En Vadikkai                  1990   

       Movie_box_office_revenue  Movie_runtime  \
134                         NaN            NaN   
270                         NaN            NaN   
673                         NaN            NaN   
14909                       NaN            NaN   
19492                       NaN           52.0   
...                         ...            ...   
61849                       NaN            NaN   
61851                       NaN            NaN   
61855                       NaN            NaN   
61872                       NaN            NaN   
61953                       NaN            NaN   

                               Movie_languages  ...  second_role_actor  \
134        ['Silent film', 'English Language']  ...                NaN   
270        ['Silent film', 'English Language']  ...                NaN   
673                       ['English Language']  ...        Ruth Roland   
14909                       ['Czech Language']  ...        Zdenek Deyl   
19492  ['French Language', 'English Language']  ...  Jean-Pierre Gorin   
...                                        ...  ...                ...   
61849                       ['Tamil Language']  ...          Pasupathy   
61851                       ['Tamil Language']  ...          Vinay Rai   
61855                       ['Tamil Language']  ...       Gowri Munjal   
61872                      ['Telugu language']  ...                NaN   
61953                       ['Tamil Language']  ...              Dilip   

      third_role_actor  first_role_gender  second_role_gender  \
134                NaN                  F                 NaN   
270                NaN                  F                 NaN   
673                NaN                  M                   F   
14909       Eman Fiala   

In [31]:
crew_long["primaryName"] = crew_long["primaryName"].fillna(crew_long["directors"])

In [32]:
crew_long = crew_long.drop("directors", axis = 1)
crew_long = crew_long.rename(columns={"primaryName": "director"})

In [33]:
crew_long

tconst                           writers  Wikipedia movie ID  \
0      tt0000591                         nm0141150             4849466   
1      tt0000615               nm0092809,nm0533958            32986669   
2      tt0000679               nm0000875,nm0877783             5954041   
3      tt0000679               nm0000875,nm0877783             5954041   
4      tt0001101                                \N             6334750   
...          ...                               ...                 ...   
62084  tt9815072                         nm0445390            18475018   
62085  tt9834564                         nm1834334            22780907   
62086  tt9855214  nm10511503,nm10511505,nm10511504             7316179   
62087  tt9881364     nm7843279,nm0707494,nm2270180            17475562   
62088  tt9898504                                \N             8783319   

      Freebase_movie_ID                      Movie_name  Movie_release_date_x  \
0            /m/07s67rr               L'Enfant prodigue                  1907   
1            /m/04p7yxx              Robbery Under Arms                  1907   
2             /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
3             /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
4             /m/0g1kr9      Abraham Lincoln's Clemency                  1910   
...                 ...                             ...                   ...   
62084        /m/04f2nb9              Pontianak Menjerit                  2005   
62085        /m/05zwjv0              Oshidori kenkagasa                  1957   
62086         /m/0kv4t6                           Kisan                  2006   
62087        /m/04g0_80                            Gaja                  2008   
62088        /m/027jl28            Puteri Gunong Ledang                  1961   

       Movie_box_office_revenue  Movie_runtime  \
0                           NaN           90.0   
1                           NaN            NaN   
2                           NaN          120.0   
3                           NaN          120.0   
4                           NaN            NaN   
...                         ...            ...   
62084                       NaN          105.0   
62085                       NaN            NaN   
62086                       NaN            NaN   
62087                       NaN            NaN   
62088                       NaN            NaN   

                           Movie_languages               Movie_countries  ...  \
0                      ['French Language']                    ['France']  ...   
1                          ['Silent film']                 ['Australia']  ...   
2                     ['English Language']  ['United States of America']  ...   
3                     ['English Language']  ['United States of America']  ...   
4                     ['English Language']  ['United States of America']  ...   
...                                    ...                           ...  ...   
62084  ['Malay Language', 'Thai Language']                  ['Malaysia']  ...   
62085                ['Japanese Language']                     ['Japan']  ...   
62086               ['Malayalam Language']                     ['India']  ...   
62087                                   []                     ['India']  ...   
62088                   ['Malay Language']                  ['Malaysia']  ...   

       second_role_actor     third_role_actor  first_role_gender  \
0           Henri Gouget  Christiane Mandelys                  M   
1        George Merriman           Lance Vane                  M   
2            Frank Burns     George E. Wilson                  M   
3            Frank Burns     George E. Wilson                  M   
4                    NaN                  NaN                  M   
...                  ...                  ...                ...   
62084  Jalaluddin Hassan       Faizal Hussein                  F   
62085      Hibari Misora      

Now, we find the directors gender. We first try the wikidata database. 

In [38]:

sparql = SPARQLWrapper("https://query.wikidata.org/sparql")

logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')

all_data = []
chunk_size = 10000
offset = 0

while True:
    print(f"Fetching data with OFFSET {offset}...")
    try:
        chunk_data = fetch_gender_wikidata(offset, sparql, chunk_size)
        
        if not chunk_data:  
            break
        
        all_data.extend(chunk_data)
        offset += chunk_size
    
    except Exception as e:
        logging.error(f"Error fetching data at OFFSET {offset}: {e}")
        offset += chunk_size  

film_director_gender = pd.DataFrame(all_data)
print("Data fetching complete.")


Fetching data with OFFSET 0...
HTTP 429: Retrying in 2.00 seconds...
Fetching data with OFFSET 10000...
Fetching data with OFFSET 20000...
Fetching data with OFFSET 30000...
Fetching data with OFFSET 40000...
Fetching data with OFFSET 50000...
Fetching data with OFFSET 60000...
Fetching data with OFFSET 70000...
Fetching data with OFFSET 80000...
HTTP 429: Retrying in 2.00 seconds...
Fetching data with OFFSET 90000...
Fetching data with OFFSET 100000...


In [37]:
film_director_gender = film_director_gender.rename(columns={"filmDirector": "director", "gender": "gender_directors"})

Let's now add the gender column in our dataset

In [38]:
crew_long = crew_long.merge(film_director_gender, on="director",  how="left")

In [39]:
crew_long = crew_long.rename(columns={"gender": "gender_actors"})

In [40]:
crew_long.columns

Index(['tconst', 'writers', 'Wikipedia movie ID', 'Freebase_movie_ID',
       'Movie_name', 'Movie_release_date_x', 'Movie_box_office_revenue',
       'Movie_runtime', 'Movie_languages', 'Movie_countries', 'Movie_genres',
       'isAdult', 'Movie_release_date_y', 'genres', 'gender_actors', 'actors',
       'first_role_actor', 'second_role_actor', 'third_role_actor',
       'first_role_gender', 'second_role_gender', 'third_role_gender', 'F',
       'M', 'director', 'birthYear', 'deathYear', 'gender_directors'],
      dtype='object')

Were we able to retrieve the gender for every director?

In [41]:
crew_long.gender_directors.isna().sum()

20307

We are still missing 20307 genders. We expect this number to be a bit smaller because we have duplicates in the data.

In [42]:
non_attributed = crew_long[crew_long.gender_directors.isnull()]

In [43]:
non_attributed = non_attributed.drop_duplicates(subset=["director"])

We now retrieve the missing genders. The approach is as follows:

- For each director, query wikipedia in the french language and look at the profession section. If we find male words ("producteur", "réalisateur") then the director is a man. If we find female words ("productrice", "réalisatrice") then the director is a woman.
- If the french wikipedia page does not exist for a given director, or if we are missing the keywords, then we fall back on the english wikipedia and calculate the proportion of male words (He, him, his, himself) vs the proportoin of female words (Her, hers, herself). 
- If the english wikipedia page does not exist or we reach a tie in the proportion of words, we try the gender-guesser library. 
- If the gender-guesser library fails, or decides that a name is androgenous, then we use Namsor, an API that guesses the gender. This is our last resort because the API requires tokens and each account has a limited number of them. We thus created an acccount and used it. 

**If you wish to run this code, please create an account on Namsor (https://namsor.app/) and put your API Key**

In [ ]:
NAMSOR_API_KEY = "Insert API key here"

In [72]:
n_cores = 8  
non_attributed = process_gender_parallel(non_attributed, NAMSOR_API_KEY=NAMSOR_API_KEY, n_cores=n_cores)
print("Gender assignment for sample completed")

    

Processing: 100%|██████████| 8250/8250 [11:41<00:00, 11.76it/s]


Gender assignment for sample completed and saved.


In [74]:
non_attributed.Gender.value_counts()

Gender
male      7480
female     770
Name: count, dtype: int64

In [21]:
non_attributed["Gender Source"].value_counts()

Gender Source
Gender Guesser       3002
Wikipedia English    1845
NamSor               1645
Wikipedia French     1609
Name: count, dtype: int64

In [77]:
non_attributed_cleaned = non_attributed[["director", "Gender"]]

We merge the directors with the gender

In [275]:
crew_merged = crew_long.merge(non_attributed_cleaned, on="director", how="left")

In [276]:
crew_merged

tconst                           writers  Wikipedia movie ID  \
0      tt0000591                         nm0141150             4849466   
1      tt0000615               nm0092809,nm0533958            32986669   
2      tt0000679               nm0000875,nm0877783             5954041   
3      tt0000679               nm0000875,nm0877783             5954041   
4      tt0001101                                \N             6334750   
...          ...                               ...                 ...   
70980  tt9815072                         nm0445390            18475018   
70981  tt9834564                         nm1834334            22780907   
70982  tt9855214  nm10511503,nm10511505,nm10511504             7316179   
70983  tt9881364     nm7843279,nm0707494,nm2270180            17475562   
70984  tt9898504                                \N             8783319   

      Freebase_movie_ID                      Movie_name  Movie_release_date_x  \
0            /m/07s67rr               L'Enfant prodigue                  1907   
1            /m/04p7yxx              Robbery Under Arms                  1907   
2             /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
3             /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
4             /m/0g1kr9      Abraham Lincoln's Clemency                  1910   
...                 ...                             ...                   ...   
70980        /m/04f2nb9              Pontianak Menjerit                  2005   
70981        /m/05zwjv0              Oshidori kenkagasa                  1957   
70982         /m/0kv4t6                           Kisan                  2006   
70983        /m/04g0_80                            Gaja                  2008   
70984        /m/027jl28            Puteri Gunong Ledang                  1961   

       Movie_box_office_revenue  Movie_runtime  \
0                           NaN           90.0   
1                           NaN            NaN   
2                           NaN          120.0   
3                           NaN          120.0   
4                           NaN            NaN   
...                         ...            ...   
70980                       NaN          105.0   
70981                       NaN            NaN   
70982                       NaN            NaN   
70983                       NaN            NaN   
70984                       NaN            NaN   

                           Movie_languages               Movie_countries  ...  \
0                      ['French Language']                    ['France']  ...   
1                          ['Silent film']                 ['Australia']  ...   
2                     ['English Language']  ['United States of America']  ...   
3                     ['English Language']  ['United States of America']  ...   
4                     ['English Language']  ['United States of America']  ...   
...                                    ...                           ...  ...   
70980  ['Malay Language', 'Thai Language']                  ['Malaysia']  ...   
70981                ['Japanese Language']                     ['Japan']  ...   
70982               ['Malayalam Language']                     ['India']  ...   
70983                                   []                     ['India']  ...   
70984                   ['Malay Language']                  ['Malaysia']  ...   

      first_role_gender  second_role_gender  third_role_gender  F  M  \
0                     M                   M                  F  1  2   
1                     M                   M                  M  0  3   
2                     M                   M                  M  0  3   
3                     M                   M                  M  0  3   
4                     M                 NaN                NaN  0  1   
...                 ...                 ...                ... .. ..   
70980                 F                   M                  M  1  2  

In [277]:
crew_merged.columns

Index(['tconst', 'writers', 'Wikipedia movie ID', 'Freebase_movie_ID',
       'Movie_name', 'Movie_release_date_x', 'Movie_box_office_revenue',
       'Movie_runtime', 'Movie_languages', 'Movie_countries', 'Movie_genres',
       'isAdult', 'Movie_release_date_y', 'genres', 'gender_actors', 'actors',
       'first_role_actor', 'second_role_actor', 'third_role_actor',
       'first_role_gender', 'second_role_gender', 'third_role_gender', 'F',
       'M', 'director', 'birthYear', 'deathYear', 'gender_directors',
       'Gender'],
      dtype='object')

In [278]:
crew_merged['gender_directors'] = crew_merged['gender_directors'].fillna(crew_merged['Gender'])    

In [279]:
crew_merged = crew_merged.drop("Gender", axis=1)
crew_merged

tconst                           writers  Wikipedia movie ID  \
0      tt0000591                         nm0141150             4849466   
1      tt0000615               nm0092809,nm0533958            32986669   
2      tt0000679               nm0000875,nm0877783             5954041   
3      tt0000679               nm0000875,nm0877783             5954041   
4      tt0001101                                \N             6334750   
...          ...                               ...                 ...   
70980  tt9815072                         nm0445390            18475018   
70981  tt9834564                         nm1834334            22780907   
70982  tt9855214  nm10511503,nm10511505,nm10511504             7316179   
70983  tt9881364     nm7843279,nm0707494,nm2270180            17475562   
70984  tt9898504                                \N             8783319   

      Freebase_movie_ID                      Movie_name  Movie_release_date_x  \
0            /m/07s67rr               L'Enfant prodigue                  1907   
1            /m/04p7yxx              Robbery Under Arms                  1907   
2             /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
3             /m/0fgfyy  The Fairylogue and Radio-Plays                  1908   
4             /m/0g1kr9      Abraham Lincoln's Clemency                  1910   
...                 ...                             ...                   ...   
70980        /m/04f2nb9              Pontianak Menjerit                  2005   
70981        /m/05zwjv0              Oshidori kenkagasa                  1957   
70982         /m/0kv4t6                           Kisan                  2006   
70983        /m/04g0_80                            Gaja                  2008   
70984        /m/027jl28            Puteri Gunong Ledang                  1961   

       Movie_box_office_revenue  Movie_runtime  \
0                           NaN           90.0   
1                           NaN            NaN   
2                           NaN          120.0   
3                           NaN          120.0   
4                           NaN            NaN   
...                         ...            ...   
70980                       NaN          105.0   
70981                       NaN            NaN   
70982                       NaN            NaN   
70983                       NaN            NaN   
70984                       NaN            NaN   

                           Movie_languages               Movie_countries  ...  \
0                      ['French Language']                    ['France']  ...   
1                          ['Silent film']                 ['Australia']  ...   
2                     ['English Language']  ['United States of America']  ...   
3                     ['English Language']  ['United States of America']  ...   
4                     ['English Language']  ['United States of America']  ...   
...                                    ...                           ...  ...   
70980  ['Malay Language', 'Thai Language']                  ['Malaysia']  ...   
70981                ['Japanese Language']                     ['Japan']  ...   
70982               ['Malayalam Language']                     ['India']  ...   
70983                                   []                     ['India']  ...   
70984                   ['Malay Language']                  ['Malaysia']  ...   

          third_role_actor  first_role_gender  second_role_gender  \
0      Christiane Mandelys                  M                   M   
1               Lance Vane                  M                   M   
2         George E. Wilson                  M                   M   
3         George E. Wilson                  M                   M   
4                      NaN                  M                 NaN   
...                    ...                ...                 ...   
70980       Faizal Hussein                  F                   M   
70981          Akio Ko

Do we still have missing genders?

In [58]:
crew_merged.gender_directors.isna().sum()

0

We have managed to retrieve all genders

In [281]:
crew_merged.columns

Index(['tconst', 'writers', 'Wikipedia movie ID', 'Freebase_movie_ID',
       'Movie_name', 'Movie_release_date_x', 'Movie_box_office_revenue',
       'Movie_runtime', 'Movie_languages', 'Movie_countries', 'Movie_genres',
       'isAdult', 'Movie_release_date_y', 'genres', 'gender_actors', 'actors',
       'first_role_actor', 'second_role_actor', 'third_role_actor',
       'first_role_gender', 'second_role_gender', 'third_role_gender', 'F',
       'M', 'director', 'birthYear', 'deathYear', 'gender_directors'],
      dtype='object')

We fix the release dates

In [286]:
crew_merged["Movie_release_date_x"] = crew_merged["Movie_release_date_x"].apply(lambda x: x if x != 0 else np.nan)

In [287]:
crew_merged["Movie_release_date_x"] = crew_merged["Movie_release_date_x"].fillna(crew_merged["Movie_release_date_y"])
crew_merged["Movie_release_date_x"].isna().sum()

0

In [288]:
crew_merged = crew_merged.drop("Movie_release_date_y", axis=1)

In [289]:
crew_merged = crew_merged.rename(columns={"Movie_release_date_x": "Movie_release_date"})

In [290]:
print(crew_merged['Movie_release_date'].unique())

[1907. 1908. 1910. 1913. 1911. 1912. 1914. 1916. 1915. 1919. 1917. 1918.
 1930. 1920. 1921. 1922. 1923. 1924. 1925. 1926. 1935. 1927. 1928. 1929.
 1931. 1939. 1932. 1933. 1951. 1934. 1948. 1937. 1936. 1941. 1947. 1938.
 1940. 2006. 1942. 1944. 1990. 1943. 1950. 1952. 2001. 1945. 1949. 1946.
 1973. 1953. 1960. 1958. 1954. 1955. 1961. 1977. 1957. 1956. 2005. 2000.
 1968. 1959. 1962. 1964. 1963. 1974. 1972. 1965. 1966. 1970. 1967. 1969.
 1980. 1979. 2008. 1981. 1978. 1971. 2004. 1989. 2003. 1975. 1976. 1982.
 2002. 1984. 1983. 1988. 2007. 1996. 1993. 1985. 1987. 1997. 1998. 1986.
 1995. 1991. 2011. 1999. 1992. 1994. 2010. 1903. 2009. 2012. 2018. 2015.
 2014. 2016. 2019. 2013.    0. 2022. 2025. 2017.]


In [291]:
crew_merged[crew_merged['Movie_release_date'] == 0]

tconst                        writers  Wikipedia movie ID  \
58877  tt0808372  nm1951398,nm2081046,nm0463620             5121174   
63025  tt1240952  nm0289696,nm3067697,nm2053173            27277059   
63026  tt1240952  nm0289696,nm3067697,nm2053173            27277059   
63027  tt1240952  nm0289696,nm3067697,nm2053173            27277059   
69845  tt2417504                      nm5291683            33561300   
70031  tt2793006            nm5587240,nm5827238            33354076   
70843  tt7612654                      nm3342905            31308475   

      Freebase_movie_ID              Movie_name  Movie_release_date  \
58877         /m/0d3n25        Metal Gear Solid                 0.0   
63025        /m/0bwkqtt     Empires of the Deep                 0.0   
63026        /m/0bwkqtt     Empires of the Deep                 0.0   
63027        /m/0bwkqtt     Empires of the Deep                 0.0   
69845        /m/0hgq0mv  498A: The Wedding Gift                 0.0   
70031        /m/0h7qyxh                 Kaptaan                 0.0   
70843        /m/0gj9zwk          Pendulum Drift                 0.0   

       Movie_box_office_revenue  Movie_runtime  \
58877                       NaN            NaN   
63025                       NaN            NaN   
63026                       NaN            NaN   
63027                       NaN            NaN   
69845                       NaN            NaN   
70031                       NaN            NaN   
70843                       NaN            NaN   

                                         Movie_languages  \
58877  ['French Language', 'Japanese Language', 'Engl...   
63025                                                 []   
63026                                                 []   
63027                                                 []   
69845                                                 []   
70031                                                 []   
70843                                                 []   

                             Movie_countries  ... third_role_actor  \
58877  ['United States of America', 'Japan']  ...              NaN   
63025                                     []  ...    Davy Williams   
63026                                     []  ...    Davy Williams   
63027                                     []  ...    Davy Williams   
69845                                     []  ...   Supriya Karnik   
70031                           ['Pakistan']  ...     Abdul Mannan   
70843                           ['Scotland']  ...  Marta Benvenuty   

       first_role_gender second_role_gender third_role_gender  F  M  \
58877                  M                NaN               NaN  0  1   
63025                  F                  F                 M  2  1   
63026                  F                  F                 M  2  1   
63027                  F                  F                 M  2  1   
69845                  M                  F                 F  2  1   
70031                  F                  F                 M  2  1   
70843                  F                  M                 F  2  1   

                  director birthYear deathYear gender_directors  
58877  Jordan Vogt-Roberts      1984        \N             male  
63025       Michael French        \N        \N             male  
63026    Jonathan Lawrence        \N        \N             male  
63027         Scott Miller        \N        \N             male  
69845        Suhaib Ilyasi        \N        \N             male  
70031     Faisal Aman Khan        \N        \N             male  
70843        Shehzad Afzal        \N        \N             male  

[7 rows x 27 columns]

In [292]:
crew_merged = crew_merged[crew_merged['Movie_release_date'] != 0]

In [293]:
crew_merged["Movie_release_date"] = crew_merged["Movie_release_date"].astype(int)

In [294]:
crew_merged["Movie_release_date"]

0        1907
1        1907
2        1908
3        1908
4        1910
         ... 
70980    2005
70981    1957
70982    2006
70983    2008
70984    1961
Name: Movie_release_date, Length: 70978, dtype: int32

We convert the release date to daytime

In [75]:
crew_merged["Movie_release_date"] = pd.to_datetime(crew_merged['Movie_release_date'], format = "%Y").dt.year

For this project, we only focus on Men and women, because we do not have enough samples about other genders

In [76]:
crew_merged = crew_merged[crew_merged['gender_directors'].isin(['male', 'female'])]

Finally, we save our dataframe into a pickle

In [ ]:
crew_merged.to_pickle("data/data/pickles/crew_gender.pkl")